src/plugins/formulas/formulas.js

```diff
@@ -126,318 +157,400 @@ export class Formulas extends BasePlugin {
   }
 
   /**
-   * Returns cell value (evaluated from formula expression) at specified cell coords.
-   *
-   * @param {number} row Row index.
-   * @param {number} column Column index.
-   * @returns {*}
+   * Triggered on `updateSettings`.
    */
-  getCellValue(row, column) {
-    const cell = this.sheet.getCellAt(row, column);
+  updatePlugin() {
+    const hotSettings = this.hot.getSettings();
 
-    return cell ? (cell.getError() || cell.getValue()) : void 0;
-  }
+    this.#settings = hotSettings[PLUGIN_KEY];
 
-  /**
-   * Checks if there are any formula evaluations made under specific cell coords.
-   *
-   * @param {number} row Row index.
-   * @param {number} column Column index.
-   * @returns {boolean}
-   */
-  hasComputedCellValue(row, column) {
-    return this.sheet.getCellAt(row, column) !== null;
-  }
+    if (this.#settings.engine) {
+      this.engine.updateConfig(mergeEngineSettings(this.#settings));
+    }
```

Comment:
@nikersify we've got a bug report for this code https://codesandbox.io/s/preversion3-forked-z185r

Here we will always overwrite the settings, even if the `updateSettings` does not contain any for the formula plugin. Exception will be thrown with:
```ts
hot.updateSettings({
  colWidths: function (columnIndex) {
      return 400;
  }
});
```

We should only update settings if they're are set.

```ts
if (typeof hotSettings[PLUGIN_KEY] === 'undefined') {
  return;
}
```

Also for some reason the `#settings.engine` exists in this use case scenario. We should investigate

Reply:
Work continues at https://github.com/handsontable/handsontable/pull/8001